### DISCUSSION TITLE
Empatica E4 feature extraction and ML

### DISCUSSION THREAD

**Discussion by hidaIgo** [neuropsychology/NeuroKit#981] [Q&A]
I am trying to extract features from the empatica E4 device. My events last 1 hour, 3 or 4 per day and 10 days in total.

The main problem I am having is in segmentation/feature extraction to get the final dataset to input to a ML model

1. I am not sure if grouping the events (1 row by 1 event) with interval related analysis (I think) is the right way to introduce the data to the model, because if I have 30 events, there will only be 30 rows of data. Do you have any example training a model?

2. The empatica E4 sensors have different frequency, how should I deal with this to apply nk functions?

Thank you

**Comment by DominiqueMakowski** [comment]
1. Yes for ML you might probably want different features (or feed the whole epoched signal) but that's really not my area so I can't help much more there
2. Resample all to the highest is probably the easiest route, or process the signals separately
